rustls-platform-verifier/src/verification/windows.rs

```diff
@@ -521,7 +565,24 @@ impl Verifier {
             .chain(Some(0))
             .collect();
 
-        let cert_chain = store.new_chain_in(&primary_cert, now)?;
+        let mut cert_chain = store.new_chain_in(&primary_cert, now)?;
+
+        // If we have extra roots and the building the chain gave us an error, we try to build a
+        // new one with the extra roots.
+        if !self.extra_roots.is_empty()
+            // SAFETY: The pointer is guaranteed to be non-null.
+            && unsafe { *cert_chain.inner.as_ptr() }
+                .TrustStatus
```

Comment:
Please leave a comment around here to say we only use `TrustStatus` here because it hasn't had verification performed on it. I think its worthwhile to callout why we use `TrustStatus` up here but then use `dwError` instead after the chain verification.

Reply:
I've added a comment trying to explicit that fact